<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1388</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QSlider" name="epoch_sld">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>20</y>
     <width>160</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <number>1</number>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
   <property name="value">
    <number>6</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="epoch_lbl">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>20</y>
     <width>47</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>6</string>
   </property>
  </widget>
  <widget class="QLabel" name="fold_lbl">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>50</y>
     <width>47</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>5</string>
   </property>
  </widget>
  <widget class="QSlider" name="fold_sld">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>50</y>
     <width>160</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <number>2</number>
   </property>
   <property name="maximum">
    <number>10</number>
   </property>
   <property name="value">
    <number>5</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="batch_lbl">
   <property name="geometry">
    <rect>
     <x>950</x>
     <y>80</y>
     <width>47</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>32</string>
   </property>
  </widget>
  <widget class="QSlider" name="batch_sld">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>80</y>
     <width>160</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <number>0</number>
   </property>
   <property name="maximum">
    <number>12</number>
   </property>
   <property name="value">
    <number>5</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>260</y>
     <width>841</width>
     <height>431</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="mplvl">
    <item>
     <widget class="QWidget" name="mplwindow" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>720</x>
     <y>20</y>
     <width>47</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>EPOCHS</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>666</x>
     <y>50</y>
     <width>101</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>NUMBER_OF_FOLDS</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>700</x>
     <y>80</y>
     <width>71</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>BATCH_SIZE</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="train_low_in">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>20</y>
     <width>113</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>0.1</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>20</y>
     <width>47</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>train_low</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="train_high_in">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>50</y>
     <width>113</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>2</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>50</y>
     <width>61</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>train_high</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="plot_low_in">
   <property name="geometry">
    <rect>
     <x>530</x>
     <y>20</y>
     <width>113</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>0</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>20</y>
     <width>47</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>plot_low</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="plot_high_in">
   <property name="geometry">
    <rect>
     <x>530</x>
     <y>50</y>
     <width>113</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>4</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>50</y>
     <width>47</width>
     <height>14</height>
    </rect>
   </property>
   <property name="text">
    <string>plot_high</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_9">
   <property name="geometry">
    <rect>
     <x>200</x>
     <y>80</y>
     <width>121</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>NUMBER_OF_SAMPLES</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="samples_in">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>80</y>
     <width>113</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>20000</string>
   </property>
  </widget>
  <widget class="QLabel" name="label10">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>80</y>
     <width>81</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>PLOT_SAMPLES</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="plot_samples_in">
   <property name="geometry">
    <rect>
     <x>530</x>
     <y>80</y>
     <width>113</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>200000</string>
   </property>
  </widget>
  <widget class="QPushButton" name="reset_btn">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>120</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Reset Default</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>141</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Amirmahdi Namjoo</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_10">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>700</y>
     <width>321</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string> AI :: Computer Engineering @ Sharif University of Technology</string>
   </property>
  </widget>
  <widget class="QPushButton" name="make_btn">
   <property name="geometry">
    <rect>
     <x>450</x>
     <y>120</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Make Dataset</string>
   </property>
  </widget>
  <widget class="QPushButton" name="learn_btn">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>120</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Learn</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="function_text">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>160</y>
     <width>271</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>np.sin(x)</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>210</x>
     <y>160</y>
     <width>251</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Single variable (x) function as numpy valid input</string>
   </property>
  </widget>
  <widget class="QPushButton" name="set_btn">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>120</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Set Values</string>
   </property>
  </widget>
  <widget class="QPushButton" name="plot_btn">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>120</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Plot</string>
   </property>
  </widget>
  <widget class="QLabel" name="avg_loss_lbl">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>240</y>
     <width>151</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>0.0000</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_15">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>240</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Average Loss:</string>
   </property>
  </widget>
  <widget class="QLabel" name="best_loss_lbl">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>240</y>
     <width>131</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>0.0000</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>240</y>
     <width>61</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>Best Loss:</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="pred_x_in">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>190</y>
     <width>113</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>0.00</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>190</y>
     <width>71</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Prediction x:</string>
   </property>
  </widget>
  <widget class="QPushButton" name="predict_btn">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>190</y>
     <width>91</width>
     <height>23</height>
    </rect>
   </property>
   <property name="text">
    <string>Predict</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_14">
   <property name="geometry">
    <rect>
     <x>370</x>
     <y>210</y>
     <width>91</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>Prediction result:</string>
   </property>
  </widget>
  <widget class="QLabel" name="predict_y">
   <property name="geometry">
    <rect>
     <x>470</x>
     <y>210</y>
     <width>231</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>0.0000</string>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>1000</x>
     <y>30</y>
     <width>371</width>
     <height>661</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="log_lv">
    <item>
     <widget class="QWidget" name="log_text" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
